### PR TITLE
[GPU] fix a bug for resample_pil_ref/sdpa_micro

### PR DESCRIPTION
**Pull request by sungeunk**
**resample_pil_ref**
- wrong size for internal buffer
- wrong internal buffer order

**sdpa_micro**
- add INPUTx_OFFSET for Q/K/V.

### Tickets:
 - 149933

### CODE REVIEW

File: src/plugins/intel_gpu/src/kernel_selector/cl_kernels/sdpa_micro.cl [hunk 1]
```diff
@@ -183,9 +183,9 @@ KERNEL(micro_sdpa)(OPTIONAL_SHAPE_INFO_ARG
     const bool need_sum_barrier = (ugemm_vs_barrier_count == 0);
 
     /* Locate K/Q/V/A matrices within batch */
-    K += KEY_OFF(b1, (b0 / KV_GROUP_SIZE), 0, 0);
-    Q += QRY_OFF(b1, b0, 0, 0);
-    V += VAL_OFF(b1, (b0 / KV_GROUP_SIZE), 0, 0);
+    K += KEY_OFF(b1, (b0 / KV_GROUP_SIZE), 0, 0) + INPUT1_OFFSET;
+    Q += QRY_OFF(b1, b0, 0, 0) + INPUT0_OFFSET;
+    V += VAL_OFF(b1, (b0 / KV_GROUP_SIZE), 0, 0) + INPUT2_OFFSET;
     A += DST_OFF(b1, b0, 0, 0, 0);
```
Comment (isanghao) [inline src/plugins/intel_gpu/src/kernel_selector/cl_kernels/sdpa_micro.cl:189]: could you add test for this case?

File: src/plugins/intel_gpu/tests/functional/subgraph_tests/sdpa.cpp [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+#include "openvino/op/group_conv.hpp"
+#include "openvino/op/multiply.hpp"
+#include "openvino/op/split.hpp"
+
+#include "openvino/opsets/opset13.hpp"
+#include "transformations/op_conversions/scaled_dot_product_attention_decomposition.hpp"
+#include "openvino/pass/manager.hpp"
+
+namespace {
+// Validate scenario where a single Constant has multiple users (like one constant is used for Convolution, ConvolutionBackpropData, Multiply, etc.)
```
Comment (vladimir-paramuzov) [inline src/plugins/intel_gpu/tests/functional/subgraph_tests/sdpa.cpp:22]: This comment seems to be irrelevant

**Review by vladimir-paramuzov — APPROVED** [review]
Overall, LGTM

File: src/plugins/intel_gpu/tests/functional/subgraph_tests/sdpa.cpp (hunk 2 above)
Comment (sungeunk) [reply src/plugins/intel_gpu/tests/functional/subgraph_tests/sdpa.cpp:22]: I did not update the comment while copying the test. I'll update

File: src/plugins/intel_gpu/src/kernel_selector/kernels/resample/resample_kernel_pil_ref.cpp [hunk 3]
```diff
@@ -52,18 +52,38 @@ Tensor::Dim ExtractDim(const DataTensor& tensor, InterpolateAxis axis) {
...
+}
+
+size_t getInputVerticalSize(const resample_params& params, bool with_pad = false) {
+    auto inputVerticalSize = ExtractDim(params.inputs[0], params.axes[eVertical]).v;
+    if (with_pad) {
+        inputVerticalSize += params.pads_begin[ConvertToNCHW(params.axes[eVertical])];
+        inputVerticalSize += params.pads_end[ConvertToNCHW(params.axes[eVertical])];
+    }
+    return inputVerticalSize;
+}
```
Comment (isanghao) [inline src/plugins/intel_gpu/src/kernel_selector/kernels/resample/resample_kernel_pil_ref.cpp:71]: is there a case where `with_pad = false` for `getInputHorizontalSize` or `getInputVerticalSize`? If not, I think we can simplify this function.

**Review by p-durandin — APPROVED** [review]
